### PR TITLE
Adding support for gateset selection in IonQ

### PR DESCRIPTION
**Pull request by ricardo-espinoza**

### CODE REVIEW

File: azure-quantum/azure/quantum/qiskit/backends/ionq.py [hunk 1]
```diff
@@ -63,6 +63,12 @@ def _translate_input(self, circuit, data_format, input_params):
         else:
             return super()._translate_input(circuit, data_format, input_params)
 
+    def _set_gateset(self, **kwargs):
+        self._gateset = kwargs.get("gateset", "qis")
```
Comment (anpaz) [inline azure-quantum/azure/quantum/qiskit/backends/ionq.py:67]:
```suggestion
        self._gateset = GETSET_MAP[kwargs.get("gateset", "qis")]
```

File: azure-quantum/azure/quantum/qiskit/backends/ionq.py [hunk 2]
```diff
@@ -92,7 +100,7 @@ def __init__(
                 "local": False,
                 "coupling_map": None,
                 "description": "IonQ simulator on Azure Quantum",
-                "basis_gates": ionq_basis_gates,
+                "basis_gates": GATESET_MAP[self.gateset()],
```
Comment (anpaz) [inline azure-quantum/azure/quantum/qiskit/backends/ionq.py:103]:
```suggestion
                "basis_gates": self.gateset(),
```

File: azure-quantum/azure/quantum/qiskit/backends/ionq.py [hunk 3]
```diff
@@ -126,7 +135,7 @@ def __init__(
                 "local": False,
                 "coupling_map": None,
                 "description": "IonQ QPU on Azure Quantum",
-                "basis_gates": ionq_basis_gates,
+                "basis_gates": GATESET_MAP[self.gateset()],
```
Comment (anpaz) [inline azure-quantum/azure/quantum/qiskit/backends/ionq.py:138]:
```suggestion
                "basis_gates": self.gateset(),
```

**Review by anpaz — COMMENTED** [review]
I would have `gateset()` return the actual gate set, not just the id, as put in the review.

File: azure-quantum/azure/quantum/qiskit/backends/ionq.py (hunk 1 above)
Comment (ricardo-espinoza) [reply azure-quantum/azure/quantum/qiskit/backends/ionq.py:67]: We discussed this offline, and we're keeping IonQ's convention.

File: azure-quantum/azure/quantum/qiskit/backends/ionq.py (hunk 3 above)
Comment (ricardo-espinoza) [reply azure-quantum/azure/quantum/qiskit/backends/ionq.py:138]: We discussed this offline, and we're keeping IonQ's convention.

File: azure-quantum/azure/quantum/qiskit/backends/ionq.py (hunk 2 above)
Comment (ricardo-espinoza) [reply azure-quantum/azure/quantum/qiskit/backends/ionq.py:103]: We discussed this offline, and we're keeping IonQ's convention.

**Comment by ricardo-espinoza** [comment]
> I would have `gateset()` return the actual gate set, not just the id, as put in the review.

Totally agree. Just aligning with the convention [here](https://github.com/Qiskit-Partners/qiskit-ionq/blob/main/qiskit_ionq/ionq_backend.py#L338).

**Pull request merged**